lib/improver/tests/set_up_test_cubes.py

```diff
@@ -446,7 +446,8 @@ def add_coordinate(incube, coord_points, coord_name, coord_units=None,
                      units=coord_units, attributes=attributes))
 
         # recalculate forecast period if time or frt have been updated
-        if is_datetime and "time" in coord_name:
+        if ("time" in coord_name and coord_units is not None
```

Comment:
Do any of the tests for set_up_test_cubes need updating?

Reply:
This change was so that a time or forecast reference time coordinate could be added using integer points values rather than datetime instances, and the forecast period would be correctly recalculated.  I've added a test showing this now happens correctly, and changed the call in places where unit tests were adding a new forecast period manually.

Reply:
Thanks for tidying this up